### PR TITLE
A couple of minor improvements

### PR DESCRIPTION
**Pull request by mjuniper**
Just reviewing the codebase to get my head into it and I noticed a couple of things:
1. Removed unnecessary closure setting up the page queue.
2. Short circuited the loop where we look for the OID field.
3. Reordered a couple of statements in the metadata function for clarity.
4. Removed unnecessary variable in metadata function.

### CODE REVIEW

**Comment by dmfenton** [comment]
Sweet, thanks @mjuniper I'll test this locally and make sure everything is good and that we have good coverage on what you touched.

File: index.js [hunk 1]
```diff
@@ -35,10 +35,8 @@ var FeatureService = function (url, options) {
   this.timeOut = 1.5 * 60 * 1000
   var concurrency = this.url.split('//')[1].match(/^service/) ? 16 : 4
 
-  // an async for requesting pages of data
-  this.pageQueue = queue(function (task, callback) {
-    this._requestFeatures(task, callback)
-  }.bind(this), concurrency)
```
Comment (dmfenton) [inline index.js:41]: turns out this binding is necessary, removing it causes 

self.timeOut at https://github.com/koopjs/featureservice/blob/master/index.js#L427 to resolve as null

Comment (mjuniper) [inline index.js:41]: Thanks for catching that. I wonder if 
`this.pageQueue = queue(this._requestFeatures.bind(this), concurrency)`
would work. That would still remove the unnecessary closure while preserving context.

**Comment by ungoldman** [comment]
This needs a rebase :wave:

**Pull request closed without merging**